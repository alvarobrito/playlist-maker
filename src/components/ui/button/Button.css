.btn {
  cursor: pointer;
  line-height: normal;
  display: inline-block;
  text-decoration: none;
  font-size: 1rem;
  padding: .6rem;
	background-color: white;
	border-radius: 10px;
  border-width: 1px;
  border-style: solid;
  text-align: center;
}

.btn--primary {
	border-color: #3b55e6;
  color: #3b55e6;
}
