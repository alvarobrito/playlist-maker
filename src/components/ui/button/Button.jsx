import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './Button.css';

const Button = ({ primary, children, ...props }) =>
  <button
    className={classnames('btn', { 'btn--primary': !!primary })}
    type="button"
    {...props}
  >
    {children}
  </button>;

Button.propTypes = {
  primary: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Button.defaultProps = {
  primary: '',
};

export default Button;
