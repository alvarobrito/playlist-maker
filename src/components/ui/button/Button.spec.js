import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button component', () => {
  const buttonText = 'Text button';

  it('snapshot', () => {
    const tree = renderer.create(
      <Button>{buttonText}</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain a text passed as children', () => {
    const wrapper = shallow(
      <Button>{buttonText}</Button>
    );
    expect(wrapper.text()).toEqual(buttonText);
  });

  it('should contain an empty render if a text is not passed as children', () => {
    const wrapper = shallow(
      <Button />
    );
    expect(wrapper.isEmptyRender()).toBe(true);
  });
});
