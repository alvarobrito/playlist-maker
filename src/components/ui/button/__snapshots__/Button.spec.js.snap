// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
<button
  className="btn"
  type="button"
>
  Text button
</button>
`;
